### PR TITLE
scripts: Remove unused variables and imports from all Python scripts

### PR DESCRIPTION
**Pull request by ulfalizer**
```
scripts: Remove unused variables in all Python scripts

Discovered with pylint3.

Use the placeholder name '_' for unproblematic unused variables. It's
what I'm used to, and pylint knows not to flag it.

Python tip:

    for i in range(n):
        some_list.append(0)

can be replaced with

    some_list += n*[0]

Similarly, 3*'\t' gives '\t\t\t'.

(Relevant here because pylint flagged the loop index as unused.)

To do integer division in Python 3, use // instead of /.
```

```
scripts: Remove unused imports in all Python scripts

Upstream open-amp PR: https://github.com/OpenAMP/open-amp/pull/168
    
Discovered with pylint3.
```

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by mbolivar — APPROVED** [review]
+1 to the runner change (`scripts/west_commands/runners/bossac.py`)

@ulfalizer if we had a flake8 control script as part of CI in the zephyr repository, just like you and I once discussed doing for west (and which we have since done in west), cleanups like this would not be necessary.

Are you going to add one?

**Comment by ulfalizer** [comment]
@mbolivar 
Working on running `pylint3` in CI with just uncontroversial stuff like this enabled, and failing if anything is found.

Could turn on more stuff as we go. I'm not big on making it too strict style-wise though.

**Comment by ulfalizer** [comment]
Had missed a bunch of unused `from ... import ...`. Fixed now.

**Review by dbkinder — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

File: boards/xtensa/intel_s1000_crb/support/create_board_img.py [hunk 1]
```diff
@@ -188,9 +188,8 @@ def main():
     ipc_load_fw(SRAM_SIZE, FLASH_PART_TABLE_OFFSET)
 
     # pad zeros until FLASH_PART_TABLE_OFFSET
-    num_zero_pad = (FLASH_PART_TABLE_OFFSET / 4) - len(flash_content)
-    for x in range(int(num_zero_pad)):
-        flash_content.append(0)
+    num_zero_pad = (FLASH_PART_TABLE_OFFSET // 4) - len(flash_content)
+    flash_content += num_zero_pad*[0]
```
Comment (pfalcon) [inline boards/xtensa/intel_s1000_crb/support/create_board_img.py]: This violates PEP8 (spaced around binary ops).

File: ext/lib/ipc/open-amp/open-amp/docs/img-src/gen-graph.py [hunk 2]
```diff
@@ -1,4 +1,3 @@
-from graphviz import Digraph
```
Comment (pfalcon) [inline ext/lib/ipc/open-amp/open-amp/docs/img-src/gen-graph.py]: This patches file in ext/.

**Review by pfalcon — CHANGES_REQUESTED** [review]
It's somewhat ironical that this improves code cleanness by such advanced means as replacing unused vars with `_` (which is Python adhoc idiom), but introduces syntax like `foo*bar`, which is bad codestyle in almost any language out there (spaces are suggested around binary operators). Please fix that, and also update samples in the commit message, to not show bad example ;-).

File: boards/xtensa/intel_s1000_crb/support/create_board_img.py (hunk 1 above)
Comment (pfalcon) [reply boards/xtensa/intel_s1000_crb/support/create_board_img.py]: Also please use `num_zero_pad * [0]`, which should clarify what happens there.

Comment (ulfalizer) [reply boards/xtensa/intel_s1000_crb/support/create_board_img.py]: No such rule in PEP 8 (not a PEP 8 fanatic either though :). They have this example:

Skärmbild från 2019-03-20 11-18-41

Most common style I've seen in all languages no space around `*` and `/`, and spaces around `+` and `-`. Doing this looks a bit ugly and inconsistent to me:

```
a = b*c + d
z = e * f
```

**Comment by ulfalizer** [comment]
> It's somewhat ironical that this improves code cleanness by such advanced means as replacing unused vars with `_` (which is Python adhoc idiom)

Seen this in Haskell too. Nice in that you can immediately that that something isn't used.

File: boards/xtensa/intel_s1000_crb/support/create_board_img.py (hunk 1 above)
Comment (ulfalizer) [reply boards/xtensa/intel_s1000_crb/support/create_board_img.py]: Got this too, so they're probably not against no spaces `*`:

Skärmbild från 2019-03-20 11-25-53

Comment (pfalcon) [reply boards/xtensa/intel_s1000_crb/support/create_board_img.py]: > No such rule in PEP 8

LOL, indeed. On a very thorough reading of https://www.python.org/dev/peps/pep-0008/#other-recommendations , 2nd bullet point:

> Always surround these binary operators with a single space on either side: assignment (=), augmented assignment (+=, -= etc.), comparisons (==, <, >, !=, <>, <=, >=, in, not in, is, is not), Booleans (and, or, not).

they don't speak about *all* binary operators, just *some*. And that's the same people which limit line size to orthodox and counter-productive 79 chars.

Well, anyway: I'll remain -1 until this is fixed, in Zephyr Python code, we have consistency re: that. Thanks.

**Comment by pfalcon** [comment]
> Seen this in Haskell too. Nice in that you can immediately that that something isn't used.

But you can't tell *what* is unused. That said, no contention here, it's a good change. (But if it wasn't made, it wouldn't be bad either.)

**Comment by ulfalizer** [comment]
> > Seen this in Haskell too. Nice in that you can immediately that that something isn't used.
> 
> But you can't tell _what_ is unused. That said, no contention here, it's a good change. (But if it wasn't made, it wouldn't be bad either.)

Yeah, it's a trade-off. In these cases, I think it's more helpful to just say "don't worry about this".

File: boards/xtensa/intel_s1000_crb/support/create_board_img.py (hunk 1 above)
Comment (ulfalizer) [reply boards/xtensa/intel_s1000_crb/support/create_board_img.py]: Yeah, FWIW, I think the line length limit in Python should be around 90, with common-sense exceptions. 79 is juust a bit too tight in many cases.

So is the Zephyr style to always put spaces around all binary operators, and write things like `a * b + c * d`? I could change it to that, though it feels a bit unusual to me.

Comment (pfalcon) [reply boards/xtensa/intel_s1000_crb/support/create_board_img.py]: > So is the Zephyr style to always put spaces around all binary operators, and write things like a * b + c * d?

I personally never saw any other codestyle in projects which care about a codestyle at all. It is definitely the codestyle for Zephyr C code. And (most of?) the Python, Perl code is like that either.

Comment (ulfalizer) [reply boards/xtensa/intel_s1000_crb/support/create_board_img.py]: `a*b + c*d` is a lot easier to read for sure, but alright, changed.

Suspect it's some kind of overreaction to "these guys are writing impenetrable walls of code with zero whitespace", that turned into "let's just make them put spaces everywhere and that'll avoid that at least".

File: ext/lib/ipc/open-amp/open-amp/docs/img-src/gen-graph.py (hunk 2 above)
Comment (ulfalizer) [reply ext/lib/ipc/open-amp/open-amp/docs/img-src/gen-graph.py]: Is that strictly verboten? A bunch of other `ext/` changes got in earlier. :)

Comment (pfalcon) [reply ext/lib/ipc/open-amp/open-amp/docs/img-src/gen-graph.py]: > Is that strictly verboten? A bunch of other ext/ changes got in earlier. :)

No, just FYI. I believe we started with a requirement "if ext/ is patched, the patch should be submitted upstream and Zephyr's commit message should include reference to upstream patch", but I'd say following that lies largely on patch author's side. But I hope you'd agree that it's easy to get into situation when one guy patches something in ext/, then another guy pushes new upstream release and overwrite any changes back.

File: boards/xtensa/intel_s1000_crb/support/create_board_img.py (hunk 1 above)
Comment (pfalcon) [reply boards/xtensa/intel_s1000_crb/support/create_board_img.py]: > a*b + c*d is a lot easier to read for sure

Well, perhaps, but unlikely the same can be said for `foobarlong*another + longbarfoo*anotherlongvar`. That's the whole idea - to have simple, consistent, scalable rules which are easy to follow.

As for why PEP8 doesn't very explicitly forbid, my bet is that it's to cater for "scientist" users who indeed may love to write `a*b + c*d`. But that's interesting crowd which also loves to index arrays starting from arbitrary numbers, their recent language has stuff like `2x` (but not `x2`, in a sense that it has completely different meaning), etc. So, I'd take it as an exception, not as something to take for granted.

Thanks for changes!

**Review by pfalcon — APPROVED** [review]

**Review by andyross — APPROVED** [review]
Throwing in approval for the tiny bit of xtensa stuff that was my fault.  The rest looks great too, but remember I'm not a python guy.

File: ext/lib/ipc/open-amp/open-amp/docs/img-src/gen-graph.py (hunk 2 above)
Comment (ulfalizer) [reply ext/lib/ipc/open-amp/open-amp/docs/img-src/gen-graph.py]: Made an upstream PR: https://github.com/OpenAMP/open-amp/pull/168

**Review by agross-oss — APPROVED** [review]

**Pull request merged**